### PR TITLE
Add missing dtype check in PrecisionRecallCurve

### LINKED ISSUE #1443: Thresholds not working?
## 🐛 Bug

Using `torchmetrics.PrecisionRecallCurve` / `torchmetrics.ROC` produce an error when setting `thresholds` to something other than `None`.

### To Reproduce

```py
import torch
import torchmetrics
x = torch.zeros(5, 2)
y = torch.zeros(5, 2)
thresholds = torch.linspace(0, 1, 100)
prc = torchmetrics.PrecisionRecallCurve(task="multilabel", num_labels=2, thresholds=thresholds)
prc(x, y)
```

Output:
```
File .../torchmetrics/utilities/data.py:228, in _bincount(x, minlength)
    226         output[i] = (x == i).sum()
    227     return output
--> 228 return torch.bincount(x, minlength=minlength)

RuntimeError: "bincount_cpu" not implemented for 'Float'
```

#### Code sample

See above.

### Expected behavior

Works as documented, where a `List` or `Tensor` of thresholds can be manually provided, rather than using the autobin feature.

### Environment

- TorchMetrics version (and how you installed TM, e.g. `conda`, `pip`, build from source): conda, 0.11
- Python & PyTorch Version (e.g., 1.0): Python 3.10, PyTorch 1.13.1
- Any other relevant information such as OS (e.g., Linux): LUbuntu 22.04

### Additional context

N/A

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Fixes #1443 
Add missing dtype check on the `target` tensor for PR-curve metric, which leads to unwanted errors later in the code.

## Before submitting

- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/metrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [x] Did you write any new **necessary tests**?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: src/torchmetrics/functional/classification/precision_recall_curve.py [hunk 1]
```diff
@@ -622,7 +632,7 @@ def _multilabel_precision_recall_curve_update(
     len_t = len(thresholds)
     # num_samples x num_labels x num_thresholds
     preds_t = (preds.unsqueeze(-1) >= thresholds.unsqueeze(0).unsqueeze(0)).long()
-    unique_mapping = preds_t + 2 * target.unsqueeze(-1)
+    unique_mapping = preds_t + 2 * target.unsqueeze(-1)  # target.long().unsqueeze(-1)
```
Comment (stancld) [inline src/torchmetrics/functional/classification/precision_recall_curve.py:635]: Is the comment intented to be there?

File: src/torchmetrics/functional/text/rouge.py [hunk 2]
```diff
@@ -39,13 +41,40 @@
 ALLOWED_ACCUMULATE_VALUES = ("avg", "best")
 
 
+def _is_internet_connection() -> bool:
```
Comment (stancld) [inline src/torchmetrics/functional/text/rouge.py:44]: I assume changes come from the commit from another MR.

File: src/torchmetrics/functional/classification/precision_recall_curve.py (hunk 1 above)
Comment (SkafteNicki) [reply src/torchmetrics/functional/classification/precision_recall_curve.py:635]: right, forgot to remove :)
Thanks
```suggestion
    unique_mapping = preds_t + 2 * target.unsqueeze(-1)
```

File: src/torchmetrics/functional/text/rouge.py (hunk 2 above)
Comment (SkafteNicki) [reply src/torchmetrics/functional/text/rouge.py]: yes, my mistake :)

**Review by stancld — APPROVED** [review]

File: src/torchmetrics/functional/classification/precision_recall_curve.py [hunk 3]
```diff
@@ -334,6 +340,10 @@ def _multiclass_precision_recall_curve_tensor_validation(
         raise ValueError(
             f"Expected `preds` to have one more dimension than `target` but got {preds.ndim} and {target.ndim}"
         )
+    if target.is_floating_point():
+        raise ValueError(
+            "Expected argument `target` to be an int or long tensor, but got tensor with dtype {target.dtype}"
```
Comment (Borda) [inline src/torchmetrics/functional/classification/precision_recall_curve.py:345]:
```suggestion
            f"Expected argument `target` to be an int or long tensor, but got tensor with dtype {target.dtype}"
```

**Review by Borda — APPROVED** [review]

**Pull request merged**